sdk/src/vec.rs

```diff
@@ -82,35 +80,31 @@ impl<T: EnvValType + RawValType> Vec<T> {
     }
 
     #[inline(always)]
-    pub fn push(&self, x: T) -> Vec<T> {
+    pub fn push(&mut self, x: T) {
         let env = self.env();
         let vec = env.vec_push(self.0.as_ref().val, x.into_raw_val(env));
-        let vec: Object = vec.in_env(env).try_into().or_abort();
-        unsafe { Self::unchecked_new(vec) }
+        self.0 = vec.in_env(env).try_into().or_abort();
     }
 
     #[inline(always)]
-    pub fn pop(&self) -> Vec<T> {
+    pub fn pop(&mut self) {
         let env = self.env();
         let vec = env.vec_pop(self.0.as_ref().val);
-        let vec: Object = vec.in_env(env).try_into().or_abort();
-        unsafe { Self::unchecked_new(vec) }
+        self.0 = vec.in_env(env).try_into().or_abort();
```

Comment:
We could reduce the instructions that get written if the env crate allowed the sdk to do:
```suggestion
        self.0.0.val = vec;
```
@graydon What do you think about us exposing an unsafe set_val-like access?

Reply:
seems fine to me. I don't even really know that I'd say it should be unsafe. all the "unsafe" adornments I've been putting on this code so far is like "users stay away, SDK-only", but I don't know that the user is going to be able to mess things up too badly if they swap one val for another on a given EnvVal.

Reply:
With that in mind I think I'll make the EnvVal pub within EnvObj. But that really raises the question: if the EnvVal is public, what is EnvObj achieving....

Reply:
Addressing this another way: https://github.com/stellar/rs-stellar-contract-env/pull/39